src/types.ts

```diff
@@ -74,3 +96,149 @@ export enum compression {
   /** High compression */
   high = 3,
 }
+
+/** Penumbra API */
+export type PenumbraAPI = {
```

Comment:
I think we spoke about this before. This is fine to type out the entire API but note you are creating redundant types. You can fix the types on the functions you want exported and let those functions be the type definition. As the library continues to expand and change, it may be a hassle to keep the type definitions synced up.

My preference is to normally just let the functions define the API but if you want to do the top level definition in addition thats fine

Reply:
I guess you are doing this because you need the type information to be passed around am i correct? 

There actually is still a way to do this. You can have recursive imports with typescript and do 
something like 

```
import * as penumbra from './index'
type PenumbraApi = typeof penumbra;
```

Reply:
It's not critically important to pass the type info around.

I was mostly using the duplicate types to help me keep track of the API overview during implementation. I could remove it but they seem fine for now.

Reply:
ok, just be sure to keep this in mind moving forward as you will have to make changes in two places whenever any of these change. maybe make a note of this so someone new to the codebase is aware of that